dart/lib/src/sentry.dart

```diff
@@ -79,6 +68,25 @@ class Sentry {
     }
   }
 
+  /// This method reads available environment variables and uses them
+  /// accordingly.
+  /// To see which environment variables are available, see [EnvironmentVariables]
+  ///
+  /// The precendence of these options are also described on
+  /// https://docs.sentry.io/platforms/dart/configuration/options/
+  static void _setEnvironmentVariables(SentryOptions options) {
+    final vars = options.environmentVariables;
+    options.dsn = options.dsn ?? vars.dsn;
+
+    if (options.environment == null) {
+      var environment = options.platformChecker.environment;
+      options.environment = vars.environment ?? environment;
+    }
```

Comment:
this wouldnt be the same as:

`options.environment = options.environment ?? options.platformChecker.environment;` or it needs to be this way?

Reply:
We do have three values which we need to consider:
- `SentryOptions.environment`
- Environment variable
- The value we set based on Darts release, profile and debug mode.

So I first check if the user has set something and then look for the other two.
Does that clear it up?

Reply:
oh yeah that makes sense, it'd be possible still with this `ternary` operator but it'd be a bit complex to read, lets keep as it is